open-api/rest-catalog-open-api.yaml

```diff
@@ -2068,6 +2162,145 @@ components:
           items:
             $ref: '#/components/schemas/PartitionStatisticsFile'
 
+    PlanTask:
+      description:
+        A JSON object that contains information provided by the server,
+        to be utilized by clients for distributed planning, should be supplied
+        as is for input in PlanTable operation.
+      type: object
+
+    FileScanTask:
+      type: object
+      required:
+        - schema
+        - spec
+        - start
+        - length
+        - data-file
+      properties:
+        data-file:
+          $ref: '#/components/schemas/ContentFile'
+        partition:
+          type: object
+          additionalProperties:
+            type: string
+        size-bytes:
+          type: number
+        start:
+          type: number
+        length:
+          type: number
+        estimated-rows-count:
+          type: number
+        delete-files:
+          type: array
+          items:
+            $ref: '#/components/schemas/ContentFile'
+        schema:
+          $ref: '#/components/schemas/Schema'
+        spec:
+          $ref: '#/components/schemas/PartitionSpec'
+        residual-filter:
+          $ref: '#/components/schemas/Expression'
+
+    TypeValue:
```

Comment:
I don't think that this is necessary.

There are two places that might send typed data values:
1. The partition struct of a data file
2. The lower and upper bound maps of a data file

Both of those only allow primitive values, so this can be simpler:
```
  CountMap:
    type: object
    properties:
      keys:
        type: array
        items:
          type: integer
      values:
        type: array
        items:
          type: integer
          format: int64

  ValueMap:
    type: object
    properties:
      keys:
        type: array
        items:
          type: integer
      values:
        type: array
        items:
          $ref: '#/components/schemas/PrimitiveTypedValue'

  Struct:
    type: array
    items:
      $ref: '#/components/schemas/PrimitiveTypedValue'
```

I'm still using `PrimitiveTypedValue` rather than allowing any JSON value (which is what we do in expressions) because this must be a value suitable for `SingleValueParser`.

Reply:
comment for https://github.com/apache/iceberg/pull/9717

Reply:
have rebased on pr 9717 https://github.com/apache/iceberg/commit/bb53c3d4e0e27ac6706803c2371793ad2476ae04 

 will resolve this for now